doc/develop/application/index.rst

```diff
@@ -67,107 +67,104 @@ generated in a build directory; Zephyr does not support "in-tree" builds.
-    the west command-line tool.
-
-The Zephyr source tree also contains the following top-level
-directories, each of which may have one or more additional levels of
-subdirectories which are not described here.
+Based on where the source code of the application is located we can distinguish
+between three basic application types.
 
-:file:`arch`
-    Architecture-specific kernel and system-on-chip (SoC) code.
-    Each supported architecture (for example, x86 and ARM)
-    has its own subdirectory,
-    which contains additional subdirectories for the following areas:
+You can find out more about how the build system supports all the application
+types described in this section in the :ref:`cmake_pkg` section.
 
-    * architecture-specific kernel source files
-    * architecture-specific kernel include files for private APIs
+.. _zephyr-repo-app:
 
-:file:`soc`
-    SoC related code and configuration files.
-
-:file:`boards`
-    Board related code and configuration files.
-
-:file:`doc`
-    Zephyr technical documentation source files and tools used to
-    generate the https://docs.zephyrproject.org web content.
+Zephyr repository application
+=============================
 
-:file:`drivers`
-    Device driver code.
+The default Zephyr :ref:`west workspace <west-workspaces>` has the following
+structure:
 
-:file:`dts`
-    :ref:`devicetree <dt-guide>` source files used to describe non-discoverable
-    board-specific hardware details.
+.. code-block:: none
 
-:file:`include`
-    Include files for all public APIs, except those defined under :file:`lib`.
+   zephyrproject/
+   ├─── .west/
+   │    └─── config
+   └─── zephyr/
+        ├── arch/
+        ├── boards/
+        ├── cmake/
+        ├── samples/
+        │    ├── hello_world/
+        │    └── ...
+        ├── tests/
+        └── ...
 
-:file:`kernel`
-    Architecture-independent kernel code.
+Any application located within the ``zephyr`` folder is referred to as a Zephyr
+repository application.
+For example, the ``hello_world`` :ref:`sample <hello_world>` is a Zephyr
+repository application.
 
-:file:`lib`
-    Library code, including the minimal standard C library.
+.. _zephyr-workspace-app:
 
-:file:`misc`
-    Miscellaneous code that doesn't belong to any of the other top-level
-    directories.
+Zephyr workspace application
+============================
 
-:file:`samples`
-    Sample applications that demonstrate the use of Zephyr features.
+A Zephyr :ref:`workspace <west-workspaces>` has the following overall structure:
 
-:file:`scripts`
-    Various programs and other files used to build and test Zephyr
-    applications.
+.. code-block:: none
 
-:file:`cmake`
-    Additional build scripts needed to build Zephyr.
+   zephyrproject/
+   ├─── .west/
+   │    └─── config
+   ├─── zephyr/
+   ├─── bootloader/
+   ├─── modules/
+   ├─── tools/
+   ├─── <vendor/private-repositories>/
+   └─── applications/
+        └── app/
```

Comment:
It could, but this is irrelevant to this section, which categorizes applications based on their location. When we talk about workspace here we do not necessarily mean the default Zephyr workspace, it could also be a workspace generated by a manifest tha imports the zephyr one.